### PR TITLE
make sure arguments are tuples otherwise they won't be hashable

### PR DESCRIPTION
**Pull request by Krovatkin**
make sure arguments are tuples otherwise they won't be hashable if used in autograd.py or any other places that uses dictionaries for that matter

### CODE REVIEW

File: torchgen/model.py [hunk 1]
```diff
@@ -2056,27 +2056,30 @@ def symints_to_ints(self) -> "Arguments":
         if arguments.self_arg:
             arguments = dataclasses.replace(
                 arguments,
-                pre_self_positional=[
-                    x.symint_to_int() for x in arguments.pre_self_positional
-                ],
+                pre_self_positional=tuple(
+                    [x.symint_to_int() for x in arguments.pre_self_positional]
```
Comment (albanD) [inline torchgen/model.py:2060]: Why do you need the temporary list?

Comment (Krovatkin) [reply torchgen/model.py:2060]: Um, I've seen this pattern in quite a few places: https://github.com/pytorch/pytorch/blob/master/torch/autograd/functional.py#L671 

What should I do instead, `tuple((x for x in range(10)))` ?

Comment (albanD) [reply torchgen/model.py:2060]: Just `tuple(x for x in range(10))` no need for intermediary objects.

Comment (Krovatkin) [reply torchgen/model.py:2060]: sounds gut, I'll make a change

**Review by bdhirsh — APPROVED** [review]
lgtm

**Review by albanD — APPROVED** [review]
:+1:

**Pull request closed without merging**